crates/e2e/src/backend.rs

```diff
@@ -92,67 +103,173 @@ pub trait ContractsBackend<E: Environment> {
     type Error;
     /// Event log type.
     type EventLog;
+    /// Start building an instantiate call using a builder pattern.
+    ///
+    /// # Example
+    ///
+    /// ```ignore
+    /// // Constructor method
+    /// let mut constructor = FlipperRef::new(false);
+    /// let contract = client
+    ///     .instantiate("flipper", &ink_e2e::alice(), &mut constructor)
+    ///     // Optional arguments
+    ///     // Send 100 units with the call.
+    ///     .value(100)
+    ///     // Add 10% margin to the gas limit
+    ///     .extra_gas_portion(10)
+    ///     .storage_deposit_limit(100)
+    ///     // Submit the call for on-chain execution.
+    ///     .submit()
+    ///     .await
+    ///     .expect("instantiate failed");
+    /// ```
+    fn instantiate<'a, Contract: Clone, Args: Send + Clone + Encode + Sync, R>(
+        &'a mut self,
+        contract_name: &'a str,
+        caller: &'a Keypair,
```

Comment:
just wondering... maybe `alice()` should be moved to default as well (similarly to value, storage deposit and margin)?

Reply:
IMHO, the caller should be specified explicitly. I don't want to hide this information from the developer per se